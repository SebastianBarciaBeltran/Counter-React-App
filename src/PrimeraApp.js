// componentes en base a funciones o constantes
import React from 'react';
import PropTypes from 'prop-types';
// import React, { Fragment } from 'react';

const PrimeraApp = ({ saludo, subtitulo } ) => {
    // return <h1>Hola Mundo</h1>;
    return (
        // <Fragment>
        //     <h1>Hola Mundo</h1>
        //     <p>Mi primera aplicaci√≥n con React</p>
        // </Fragment>
        <>
            <h1> { saludo } </h1>
             {/* <pre>{ JSON.stringify(saludo, null, 3)}</pre> */}
            <p>{ subtitulo }</p> 
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
} 

PrimeraApp.defaultProps ={
    subtitulo : 'soy un subtitulo'
}

export default PrimeraApp;