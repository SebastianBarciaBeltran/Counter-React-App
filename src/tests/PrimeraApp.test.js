import '@testing-library/jest-dom';
import React       from 'react';
import { render }  from '@testing-library/react';
import { shallow } from 'enzyme';
import PrimeraApp  from '../PrimeraApp';

describe('Prueba en <PrimeraApp>', () => {

    test('Debe de mostrar <PrimeraApp /> correctamente', () => {
        // SIN ENZYME
        // const saludo = 'Hola, soy Goku';
        // const { getByText } = render( <PrimeraApp saludo={saludo} /> );
        // wrapper.getByText();
        // expect( getByText(saludo) ).toBeInTheDocument();


        // CON ENZYME
        const wrapper = shallow( <PrimeraApp saludo='Hola, soy goku' /> );

        expect( wrapper ).toMatchSnapshot();


    });

    test('Debe de mostrar el subTitulo enviado por props', () => {

        const saludo = 'Hola, soy goku';
        const subTitulo = 'soy un subtitulo'
        const wrapper = shallow( <PrimeraApp saludo={saludo} subTitulo={subTitulo} /> );

        const textoParrafo = wrapper.find('p').text();
        // console.log(textoParrafo);

        expect( textoParrafo ).toBe( subTitulo ); 

    });

});