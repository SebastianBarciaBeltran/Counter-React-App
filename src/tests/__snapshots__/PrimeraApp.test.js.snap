// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <PrimeraApp> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy goku
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;

exports[`Prueba en <PrimeraApp> Debe de mostrar el subTitulo enviado por props 1`] = `
<Fragment>
  <h1>
    Hola, soy goku
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
